flocker/provision/_libcloud.py

```diff
@@ -341,6 +344,270 @@ def _cleanup_node_named(self, name):
+
+        :param node: A libcloud node object representing the compute instance
+            of interest.
+        :return: Deferred that fires with a libcloud node object representing
+            the latest state of the instance in the case of success.
+        """
+        # Overall retry sleep time (not quite the same as timeout since
+        # it doesn't count time spent in the predicate) is just copied
+        # from the default libcloud timeout for wait_until_running.
+        # Maybe some other value would be better.
+        action = start_action(
+            action_type=u"flocker:provision:libcloud:wait_until_running",
+            name=node.name,
+            id=node.id,
+        )
+        with action.context():
+            steps = iter([15] * (600 / 15))
+            d = loop_until(
+                reactor,
+                lambda: self._async_node_in_state(
+                    reactor,
+                    node,
+                    self.TERMINAL_STATES
+                ),
+                steps=steps,
+            )
+            d = DeferredContext(d)
+
+            def got_ip_addresses():
+                d = self._async_refresh_node(reactor, node)
+                d = DeferredContext(d)
+
+                def is_running(updated_node):
+                    if updated_node.state is not NodeState.RUNNING:
+                        raise Exception("Node failed to run")
+                    return updated_node
+
+                def check_addresses(updated_node):
+                    """
+                    Check if the node has got at least one IPv4 public address
+                    and, if requested, an IPv4 private address.  If yes, then
+                    return the node object with the addresses, None otherwise.
+                    """
+                    public_ips = _filter_ipv4(updated_node.public_ips)
+                    if len(public_ips) > 0:
+                        if self._use_private_addresses:
+                            private_ips = _filter_ipv4(
+                                updated_node.private_ips
+                            )
+                            if len(private_ips) == 0:
+                                return None
+                        return updated_node
+                    else:
+                        return None
+
+                d.addCallback(is_running)
+                d.addCallback(check_addresses)
+                return d.result
+
+            # Once node is in a stable state ensure that it is running
+            # and it has necessary IP addresses assigned.
+            d.addCallback(
+                lambda _: loop_until(reactor, got_ip_addresses, steps=steps)
+            )
+
+            def failed_to_run(failure):
+                d = _retry_exception_async(reactor, node.destroy)
+                d.addCallback(lambda _: failure)
+                return d
+
+            d.addErrback(failed_to_run)
+            return d.addActionFinish()
+
```

Comment:
I think that there should be 2 blank lines between the class definition and the subsequent unrelated function. `flake8` doesn't complain here.